### PR TITLE
Add TV episode show page

### PR DESCRIPTION
**Pull request by lortza**
## Related Issues & PRs
* Follow up to PR https://github.com/mikevallano/tmdb-moviequeue/pull/253

## What This PR does
* Adds a 🔥 new page for episodes
* Allows us to see profile pics for all guest starts in addition to season cast
* Updates linking on related pages
* Adds overviews on related pages

## Screenshots
Before: no season overview. No link to episode page.
Screen Shot 2021-12-26 at 2 19 55 PM

After: Season page includes season poster, overview, and links to episode pages.
Screen Shot 2021-12-26 at 2 19 03 PM

Hey look! Episode page! It includes season cast as well as guest stars.
Screen Shot 2021-12-26 at 2 19 20 PM

## Things Learned
* It feels a bit costly to have to hit 3 endpoints for a show page, but we're not storing the parent data anywhere, so I'm not sure how to do it more efficiently just yet.
* We can consider this version the first iteration and then get clever about how to make it mo'bettah.

### CODE REVIEW

File: app/views/tmdb/tv_episode.html.erb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<h2>
```
Comment (lortza) [inline app/views/tmdb/tv_episode.html.erb:1]: Exciting new Episode show page

File: app/views/tmdb/tv_season.html.erb [hunk 2]
```diff
@@ -29,7 +27,13 @@
...
 <% @season.episodes.each do |episode| %>
   <div class='row'>
-    <h2><%= episode.episode_number %>: <%= episode.name %></h2>
+    <h2>
+      <%= link_to "#{episode.episode_number}. #{episode.name}",
+        tv_episode_path(
+          show_id: @series.show_id,
+          season_number: @season.season_number,
+          episode_number: episode.episode_number) %>
+    </h2>
```
Comment (lortza) [inline app/views/tmdb/tv_season.html.erb:36]: Link to exciting new episode page

File: config/routes.rb [hunk 3]
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
+  root 'pages#home'
```
Comment (lortza) [inline config/routes.rb:2]: Just moving the `root` to the top. :)

And adding a route to the new episode page.

File: lib/tmdb_handler.rb [hunk 4]
```diff
@@ -181,6 +181,12 @@ def tmdb_handler_tv_season(series:, show_id:, season_number:)
     )
   end
 
+  def tmdb_handler_tv_episode(show_id:, season_number:, episode_number:)
```
Comment (lortza) [inline lib/tmdb_handler.rb:184]: Get me that 'sode data, yo.

**Comment by mikevallano** [comment]
🎉 Looks good! 

Regarding the 3 requests, I took a quick look a the TMDB docs, but maybe we can quickly chat about it so I have a better understanding.

**Comment by mikevallano** [comment]
One minor thing is the encoding of names with an apostrophe. I took a quick look, but I'm not sure why.

Look at Kevin O'Grady under guest stars on the season page:
guest-stars

On the episode page:
o-grady

**Comment by lortza** [comment]
> 🎉 Looks good!
> 
> Regarding the 3 requests, I took a quick look a the TMDB docs, but maybe we can quickly chat about it so I have a better understanding.

To recap our conversation:
There are breadcrumbs on the Episode page that require 4 tiny pieces of data:
* Season name
* Season id
* Series name
* Series id

Unfortunately, the `episode` endpoint does not include that data in the response, so we have to fetch it again. ☹️ If we were persisting any of this data in our local database, it could make things easier -- or more performant. As it is, we're not sure how to make this a whole lot better without a whole lot more effort.

File: app/views/tmdb/tv_season.html.erb [hunk 5]
```diff
@@ -1,23 +1,21 @@
...
   <% if @season.cast_members.present? %>
     <h2><%= @season.name %> Cast</h2>
       <% @season.cast_members.each do |actor| %>
         <div class="headshot-container">
           <%= headshot_for(actor) %>
           <div class="name-block">
-            <p class="actor"><%= link_to "#{actor.name}", actor_search_path(actor: I18n.transliterate("#{actor.name}"))  %></p>
-            <p>As <%= truncate(actor.character_name, length: 18) %><p>
+            <p class="actor"><%= link_to "#{truncate(actor.name, length: 18, escape: false)}", actor_search_path(actor: I18n.transliterate("#{actor.name}"))  %></p>
+            <p>as <%= truncate(actor.character_name, length: 18, truncate: false, escape: false) %><p>
```
Comment (lortza) [inline app/views/tmdb/tv_season.html.erb:18]:
```suggestion
            <p>as <%= truncate(actor.character_name, length: 18, escape: false) %><p>
```

**Review by mikevallano — APPROVED** [review]
🎉

**Pull request merged**